### PR TITLE
Add basic plugin functionality to Empire

### PR DESCRIPTION
**Pull request by DakotaNelson**
Still basic, but I think it's a good start.

To test:

(open empire)
```
plugins
(some output showing one row, "example", not active)
plugin example
(some output showing it's loading)
plugins
(some output, should see stars under "loaded")
help
(output, should see option "test" listed)
test
(MAGIC)
```

Current limitations:
- no unloading functionality

### CODE REVIEW

File: lib/common/empire.py [hunk 1]
```diff
@@ -381,6 +388,59 @@ def handle_event(self, signal, sender):
...
+            print(helpers.color("[*] Plugin {} found.".format(pluginName)))
+            # note the 'plugins' package so the loader can find our plugin
+            fullPluginName = "plugins." + pluginName
+            plugin = importlib.import_module(fullPluginName)
+
+            self.load_plugin(plugin, pluginName)
+        else:
+            raise Exception("[!] Error: the plugin specified does not exist in {}.".format(pluginPath))
+
+    def load_plugin(self, module, pluginName):
```
Comment (DakotaNelson) [inline lib/common/empire.py]: this should maybe be moved to `lib/common/plugins.py`?

**Comment by DakotaNelson** [comment]
Oh yeah, this will close (the already-closed) #225.

**Comment by xorrior** [comment]
@DakotaNelson this is really cool. Working as intended, thanks again.

**Comment by DakotaNelson** [comment]
Awesome, thanks! :tada:

**Comment by DakotaNelson** [comment]
For future reference in case someone is confused and stumbles across this PR, I did a writeup on how this change works: https://strikersecurity.com/blog/empire-plugins/

**Pull request merged**